greeting-guestbook/src/operations.ts

```diff
@@ -0,0 +1,70 @@
+import {
+    TransactionContext, Transaction, CommunicatorContext, Communicator,
+    WorkflowContext, Workflow, GetApi, HandlerContext, DBOSResponseError
+} from "@dbos-inc/dbos-sdk";
+import { Knex } from "knex";
+import axios from 'axios';
```

Comment:
If we use axios, we need to add it to `dependencies` in package.json
It currently works only because in older versions we mistakenly packaged axios in our dbos-sdk.

Another option is to replace `axios` with the built-in `fetch` function, so we don't need to use any external packages. Example: https://github.com/dbos-inc/dbos-demo-apps/blob/main/shop-guide/src/utilities.ts#L92-L110

Reply:
done, please check that my `fetch` error handling is canonical

Reply:
Yeah the error handling looks good.